routes/drones.js

```diff
@@ -1,37 +1,41 @@
 const express = require('express');
 
-// require the Drone model here
+const Drone = require('../models/Drone.model')
 
 const router = express.Router();
 
-router.get('/drones', (req, res, next) => {
-  // Iteration #2: List the drones
-  // ... your code here
+router.get('/drones/', (req, res, next) => {
+  Drone.find().then((allDronesFromDb) => {
+    res.render('drones/list', {myDrones :allDronesFromDb});
+  })
+  .catch(error => console.log("Error while getting the drones from the DB: ", error));
 });
 
 router.get('/drones/create', (req, res, next) => {
-  // Iteration #3: Add a new drone
-  // ... your code here
+  res.render('drones/create-form')
 });
 
 router.post('/drones/create', (req, res, next) => {
-  // Iteration #3: Add a new drone
-  // ... your code here
+  Drone.create({name: req.body.name, propellers: req.body.propellers, maxSpeed: req.body.maxSpeed})
+  .then(()=>{res.redirect('/drones')
+  });
 });
 
 router.get('/drones/:id/edit', (req, res, next) => {
-  // Iteration #4: Update the drone
-  // ... your code here
+  Drone.findById(req.params.id)
+  .then((drone)=>{res.render('drones/update-form', drone)
+  });
 });
 
 router.post('/drones/:id/edit', (req, res, next) => {
-  // Iteration #4: Update the drone
-  // ... your code here
+  Drone.findByIdAndUpdate(req.params.id, {name: req.body.name, propellers: req.body.propellers, maxSpeed: req.body.maxSpeed})
+  .then(()=>{res.redirect('/drones') 
+  });
 });
 
-router.post('/drones/:id/delete', (req, res, next) => {
-  // Iteration #5: Delete the drone
-  // ... your code here
+router.post('/drones/:id/', (req, res, next) => {
```

Comment:
One of the [rules of CRUD](https://stackify.com/what-are-crud-operations/) naming convention is that the procedure name should end with the implemented name of the CRUD operation, you should use '/drones/:id/delete' rather then '/drones/:id/'.